### DISCUSSION TITLE
Error on build with fresh and empty template loaded

### DISCUSSION THREAD

**Discussion by SPTbranma** [Nice3point/RevitTemplates#25] [Q&A] [answered]
Hi! 

I have a problem with building the code. I tried it on different machines with VS 2022 17.2.6, .NET 4.8.04084 and RevitTemplates 2.2.0.

![image](https://user-images.githubusercontent.com/87659287/179184237-055b3a41-cc50-4c23-b232-1e12233cf835.png)

I use the modal config with installer support, tried it with GitHub and no pipeline and with Serilog on/off.

The error only appears when building and the files are all auto-generated. 

In past projects with RevitTemplates 2.0.0 i did not have and still not having this problem when building. 

Since i'm quite new to VS and C# i can not think of a working solution. I hope this is the right spot to post my problem. If I can help you with more data, please give me a quick update. 

Thank you!
Best regards,
Manuel

**Comment by Nice3point** [comment, marked as answer]
@SPTbranma  according to my tests, I don’t see any problems, I tried both VS22 and Rider
```0>------- Finished building project: RevitAddIn1. Succeeded: True. Errors: 0. Warnings: 0```

Can you send a Zip archive of the project or show the line on which the error appears?

**Reply by SPTbranma** [reply]
Of course! Thank you for the help! The ZIP is attached.

[RevitAddIn1_Build_error.zip](https://github.com/Nice3point/RevitTemplates/files/9121139/RevitAddIn1_Build_error.zip)

**Reply by Nice3point** [reply]
@SPTbranma Nuke: ```Build succeeded on 16.07.2022 12:26:51. ＼（＾ᴗ＾）／``` Build: ```Build succeeded at 12:27:19 ```
everything compiles fine, maybe the problem is in VS? invalid cache or something else, try to restore Nuget packages or write the order of actions what you are doing, you can also double click the mouse to go to the error to find out the reason

**Reply by IlyaNavodkin** [reply]
> Of course! Thank you for the help! The ZIP is attached.
> 
> [RevitAddIn1_Build_error.zip](https://github.com/Nice3point/RevitTemplates/files/9121139/RevitAddIn1_Build_error.zip)

I downloaded your file and my compilies is succeed.

**Reply by SPTbranma** [reply]
> @SPTbranma Nuke: `Build succeeded on 16.07.2022 12:26:51. ＼（＾ᴗ＾）／` Build: `Build succeeded at 12:27:19 ` everything compiles fine, maybe the problem is in VS? invalid cache or something else, try to restore Nuget packages or write the order of actions what you are doing, you can also double click the mouse to go to the error to find out the reason

Sorry for my late reply.


I've now tested it on 3 different computers, on one of which everything (VS. Templates, Framework,...) was installed for the first time, and I got the same errors everywhere. I finally tested it with Rider -> same error. I am now working again with version 2.0.0. Here everything works without problems.

Thanks anyway for your help, please let me know how I can help you if you want any files or debug logs from me.

**Reply by Nice3point** [reply]
@SPTbranma you can give a sequence of actions, for example, what settings you choose when creating a project. The best help is a screen recording from the creation of a solution to the occurrence of a problem.

**Reply by Nice3point** [reply]
I test templates on multiple machines with all possible settings before releasing on Nuget

**Reply by Nice3point** [reply]
In general, version 2.0.0 is the same as 2.2.0, there are no global differences, only a couple of useful libraries have been added, which you can see on ChangeLog

**Reply by SPTbranma** [reply]
@Nice3point - 
Yes, I think so, that you don't just test it once :-D

But I got a little further now - the problem is not with your code or template, but with the MVVM nuget. As soon as I uninstall the nuget and comment out a bit of code, everything works like clockwork. 
![image](https://user-images.githubusercontent.com/87659287/179738376-3153d14a-9e02-4b27-997f-a9bd45666c38.png)

I'm not 100% sure if I need this nuget, but without is, everything works fine now.

**Reply by SPTbranma** [reply]
![image](https://user-images.githubusercontent.com/87659287/179739390-62d8bdc9-e292-4002-bf74-1c76e8a9ebd4.png)

Yes, with the MVVM Nuget and this one and a half line edited - everything works as expected. 

![image](https://user-images.githubusercontent.com/87659287/179739671-d19d5350-9c3d-43e9-ba8b-dd3ad290c243.png)

**Reply by Nice3point** [reply]
@SPTbranma error in class duplication, where else is this attribute? More about the library here https://docs.microsoft.com/en-us/windows/communitytoolkit/mvvm/introduction

It greatly simplifies writing window programs using the MVVM pattern.

**Reply by Nice3point** [reply]
There are definitely no problems in this library, it is being developed with the support of Microsoft

**Reply by SPTbranma** [reply]
The attribute is in these two files:

- System.Diagnostics.CodeAnalysis.NotNullAttribute.cs
![image](https://user-images.githubusercontent.com/87659287/179747652-d63d4e99-ce94-4be9-a73b-9a151c08fd97.png)


- System.Diagnostics.CodeAnalysis.NotNullIfNotNullAttribute.cs
![image](https://user-images.githubusercontent.com/87659287/179747587-ce1c17b1-8a70-4650-ae4f-b451119575c1.png)


Both are generated by the Toolkit. I can confirm, when the Nuget is uninstalled, it works. As soon as i reinstall it, i get the same errors when building.

**Reply by SPTbranma** [reply]
The only thing that I, as a newbie, can see, is that in the System.Diagnostics.CodeAnalysis.NotNullIfNotNullAttribute.cs file, on line 17 the comment does not apply to the whole line. But i really don't know, if this could be the problem.

**Reply by Nice3point** [reply]
@SPTbranma The problem may be due to the version of .Net 6, which one do you have? ```dotnet --list-sdks```

**Reply by Nice3point** [reply]
https://github.com/dotnet/wpf/issues/6792#issuecomment-1183530305

**Reply by Nice3point** [reply]
I have 6.0.3 installed, they write about the problem related to the latest version, this is a .Net bug

**Reply by SPTbranma** [reply]
I have the following versions installed:
`6.0.202 [C:\Program Files\dotnet\sdk]
6.0.203 [C:\Program Files\dotnet\sdk]
6.0.302 [C:\Program Files\dotnet\sdk]
7.0.100-preview.6.22352.1 [C:\Program Files\dotnet\sdk]`

I will try the workaround and will come back to you! Thank you in the meantime!

**Reply by SPTbranma** [reply]
@Nice3point That's it!! Thank you for your time and help!

**Reply by Nice3point** [reply]
Nice) It's great that now everything works
